static/js/drafts.js

```diff
@@ -133,6 +133,13 @@ exports.restore_draft = function (draft_id) {
     if (draft.type === "stream" && draft.stream === "") {
         draft_copy.subject = "";
     }
+    if (draft.type === "stream") {
+        narrow.activate([{operand: draft_copy.stream, operator: "stream"},
+                         {operand: draft_copy.subject, operator: "topic"}]);
+    }
+    if (draft.type === "private") {
+        narrow.activate([{operand: draft_copy.reply_to, operator: "pm-with"}]);
+    }
```

Comment:
Yeah, let's make that precise:
* If the email or list of emails doesn't exist for a PM thread
* If the stream doesn't exist for a stream thread
There's no such thing as a topic not existing, so I think that list is complete?